### PR TITLE
Poc: Balloc tunable  against balloc_integration

### PR DESCRIPTION
**Pull request by nihar1313**
Balloc  changes  against balloc_intigration

### CODE REVIEW

File: balloc/ut/balloc.c [hunk 1]
```diff
@@ -112,18 +112,26 @@ int test_balloc_ut_ops(struct m0_be_ut_backend *ut_be, struct m0_be_seg *seg,
 	int                     rc;
 	time_t                  now;
 
+	struct m0_ad_balloc_format_req bcfg;
+
 	time(&now);
 	srand(now);
```
Comment (huanghua78) [inline balloc/ut/balloc.c:118]: Please remove this. They are not used.
If they are used, please use m0_rnd().

File: balloc/ut/balloc.c [hunk 2]
```diff
@@ -112,18 +112,26 @@ int test_balloc_ut_ops(struct m0_be_ut_backend *ut_be, struct m0_be_seg *seg,
...
 
+
```
Comment (huanghua78) [inline balloc/ut/balloc.c:120]: Remove this extra empty line.

File: balloc/ut/balloc.c [hunk 3]
```diff
@@ -112,18 +112,26 @@ int test_balloc_ut_ops(struct m0_be_ut_backend *ut_be, struct m0_be_seg *seg,
...
-	rc = m0_balloc_create(0, seg, grp, &motr_balloc, &M0_FID_INIT(0, 1));
+	rc = m0_balloc_create(0, seg, grp, &bcfg, &motr_balloc);
 	M0_UT_ASSERT(rc == 0);
 
 	rc = motr_balloc->cb_ballroom.ab_ops->bo_init
-		(&motr_balloc->cb_ballroom, seg, BALLOC_DEF_BLOCK_SHIFT,
-		 BALLOC_DEF_CONTAINER_SIZE, BALLOC_DEF_BLOCKS_PER_GROUP,
-		 m0_stob_ad_spares_calc(BALLOC_DEF_BLOCKS_PER_GROUP));
-
+		(&motr_balloc->cb_ballroom, seg, &bcfg);
```
Comment (huanghua78) [inline balloc/ut/balloc.c:134]:
```
	rc = motr_balloc->cb_ballroom.ab_ops->bo_init(
		&motr_balloc->cb_ballroom, seg, &bcfg);
```

Put the open "(" on the first line.

File: ioservice/storage_dev.c [hunk 4]
```diff
@@ -309,6 +313,10 @@ static void storage_devs_conf_refresh(struct m0_storage_devs *storage_devs,
 	struct m0_conf_sdev   *conf_sdev;
 	int                    rc;
 
+	struct m0_reqh_context *rctx;
+
+	rctx = m0_cs_reqh_context(reqh);
+
```
Comment (huanghua78) [inline ioservice/storage_dev.c:319]: Group the variable declaration together.

And do the initialization at the same time.
```
struct m0_reqh_context *rctx = m0_cs_reqh_context(reqh);
```

File: balloc/balloc.c [hunk 5]
```diff
@@ -547,9 +558,10 @@ static int sb_update(struct m0_balloc *bal, struct m0_sm_group *grp)
 	return M0_RC(rc);
 }
 
-static int balloc_sb_write(struct m0_balloc            *bal,
-			   struct m0_balloc_format_req *req,
-			   struct m0_sm_group          *grp)
+static int balloc_sb_write(
+			struct m0_balloc                *bal,
```
Comment (huanghua78) [inline balloc/balloc.c:562]: Why put this on a new line?

File: balloc/balloc.c [hunk 6]
```diff
@@ -3012,26 +3044,29 @@ M0_INTERNAL void m0_balloc_init(struct m0_balloc *cb)
...
-M0_INTERNAL int m0_balloc_create(uint64_t              cid,
-				 struct m0_be_seg     *seg,
-				 struct m0_sm_group   *grp,
-				 struct m0_balloc    **out,
-				 const struct m0_fid  *fid)
+M0_INTERNAL int m0_balloc_create(uint64_t			 cid,
+				 struct m0_be_seg		*seg,
+				 struct m0_sm_group		*grp,
+				 struct m0_ad_balloc_format_req	*bcfg,
+				 struct m0_balloc		**out)
```
Comment (huanghua78) [inline balloc/balloc.c:3051]: Usually, we put "TAB" at the beginning of a line, not in between chars.
Here, please remove the TABs in type and variables.

File: balloc/balloc.c [hunk 7]
```diff
@@ -3045,19 +3080,29 @@ M0_INTERNAL int m0_balloc_create(uint64_t              cid,
...
 					 &gd_btree_ops);
-			rc = balloc_trees_create(cb, &tx, fid);
+			rc = balloc_trees_create(cb, &tx, &bcfg->bfr_fid,
+						 bcfg->bfr_indexcount);
 			if (rc == 0) {
 				M0_BE_TX_CAPTURE_PTR(seg, &tx, cb);
 				*out = cb;
 			} else {
-				m0_be_btree_fini(&cb->cb_db_group_extents);
-				m0_be_btree_fini(&cb->cb_db_group_desc);
```
Comment (huanghua78) [inline balloc/balloc.c:3060]: Why is "m0_be_btree_fini(&cb->cb_db_group_desc);"  removed?

Do we need to free `cb->cb_db_group_extents'?

File: balloc/ut/balloc.c (hunk 2 above)
Comment (nihar1313) [reply balloc/ut/balloc.c:114]: Taken care, left one extra empty line between declaration and initialization

File: balloc/balloc.c (hunk 7 above)
Comment (nihar1313) [reply balloc/balloc.c:3060]: It was removed during  branch migration, added back again.

File: balloc/ut/balloc.c (hunk 1 above)
Comment (nihar1313) [reply balloc/ut/balloc.c]: Taken care..

File: balloc/ut/balloc.c (hunk 2 above)
Comment (nihar1313) [reply balloc/ut/balloc.c:114]: Done..

File: balloc/ut/balloc.c (hunk 3 above)
Comment (nihar1313) [reply balloc/ut/balloc.c]: Done..

File: ioservice/storage_dev.c (hunk 4 above)
Comment (nihar1313) [reply ioservice/storage_dev.c]: Done..

File: balloc/balloc.c (hunk 6 above)
Comment (nihar1313) [reply balloc/balloc.c]: Taken care..

**Pull request merged**